auth/service.hh

```diff
@@ -104,6 +113,30 @@ class service final : public seastar::peering_sharded_service<service> {
 
     abort_source _as;
 
+    /// `auth::service` and its contents are accessed by `service::qos::service_level_controller`
+    /// when updating the service level cache (cf. `service_level_controller::update_effective_service_levels_cache`).
+    /// However, `auth::service` is started after the node has jointed the cluster; see the commit
+    ///     562caaf6c61f75fb6e1849c70e18fcb57b6aeb46
+    /// for more details on that.
+    ///
+    /// Because we keep initialization and de-initialization paired in `main.cc`, that implies
+    /// that `auth::service` will be stopped BEFORE `service_level_controller` will be.
+    /// As a result, the cache may still be reloaded in between those two events
+    /// and effectively lead to a segmentation fault or other issues. We've already experienced
+    /// them; see issue: scylladb/scylladb#24792.
+    ///
+    /// This mutex is supposed to prevent that and that's its sole purpose.
+    ///
+    /// It's now necessary for `service_level_controller` to take this mutex whenever
+    /// it wants to reload the cache. Before accessing any member of `auth::service`,
+    /// it must make sure that the status (`_status`) of `auth::service` is equal to
+    /// `status::working`. Otherwise, it should avoid doing anything to it.
+    ///
+    /// Ditch this once it stops being necessary.
+    seastar::semaphore _sl_mutex;
```

Comment:
I think more general name like `_operation_mutex` (or other name that indicates that the service is busy) would be better. Auth doesn't depend on service levels, service levels just use the auth service and imo naming something after SL isn't right.
Theoretically, other services might be using the auth service and may want to keep it alive during their usage.

Reply:
> Theoretically, other services might be using the auth service and may want to keep it alive during their usage.

Hmm but then there'll be a queue to use auth which isn't necessary at all :/